ec-europa/eubfr-data-lake
# feat(enrich-currency): implement "enrich currency" process - EUBFR-184

# PR description

Revamps the enrichment process.

![new-archi](https://user-images.githubusercontent.com/3524389/41660805-611a77de-749d-11e8-811b-3551d0a7ce24.jpg)

## How to test

Deploy the new infrastructure in a fresh env. 

Upload a new file with some budgets in foreign currencies (check https://github.com/ec-europa/eubfr-data-lake/pull/133/files#diff-f3d895b35058d0b230e4c61c65a9fce2R6 for the list of supported currencies). Make sure there's a project end date, otherwise the record won't be enriched.

GET the record, check if the budgets have been enriched. E.g:

![enrichment](https://user-images.githubusercontent.com/3524389/41660898-ac84c7a6-749d-11e8-9ecf-9485b54de39e.jpg)
